firebase/functions/src/index.ts

```diff
@@ -343,3 +344,66 @@ exports.incProjectProgress = functions.database.ref('/v2/projects/{projectId}/re
 exports.decProjectProgress = functions.database.ref('/v2/projects/{projectId}/requiredResults/').onUpdate(() => {
     return null;
 });
+
+exports.addProjectTopicKey = functions.https.onRequest(async (_, res) => {
+    try {
+        const projectRef = await admin.database().ref('v2/projects').once('value');
+        const data = projectRef.val();
+
+        const isEmptyProject = Object.keys(data).length === 0;
+        if (isEmptyProject) {
+            res.status(404).send('No projects found');
+        }
+
+        if (!isEmptyProject && data) {
+            const newProjectData: {[key: string]: string} = {};
+
+            Object.keys(data).forEach((id) => {
+                if (data[id]?.projectTopic) {
+                    const newProjectTopic = formatProjectTopic(data[id].projectTopic);
+                    const newProjectTopicKey = newProjectTopic.toLowerCase() as string;
+                    newProjectData[`v2/projects/${id}/projectTopicKey`] = newProjectTopicKey;
+                    newProjectData[`v2/projects/${id}/projectTopic`] = newProjectTopic;
+                }
+            });
+
+            await admin.database().ref().update(newProjectData);
+            const updatedProjectsCount = Object.keys(newProjectData).length;
+            res.status(200).send(`Updated ${updatedProjectsCount} projects.`);
+        }
+    } catch (error) {
+        console.log(error);
+        res.status(500).send('Some error occurred');
+    }
+});
+
+exports.addUserNameLowercase = functions.https.onRequest(async (_, res) => {
+    try {
+        const userRef = await admin.database().ref('v2/users').once('value');
+        const data = userRef.val();
+
+        const isEmptyUser = Object.keys(data).length === 0;
+        if (isEmptyUser) {
+            res.status(404).send('No user found');
+        }
+
+        if (!isEmptyUser && data) {
+            const newUserData: {[key: string]: string} = {};
+
+            Object.keys(data).forEach((id) => {
+                if (data[id]?.username) {
+                    const usernameWithoutSpace = data[id].username.replace(/\s+/g, '');
+                    const newUserName = usernameWithoutSpace.toLowerCase() as string;
+                    newUserData[`v2/users/${id}/userName`] = newUserName;
```

Comment:
According to @frozenhelium, let's not update username but instead add usernameKey.